### PR TITLE
[Test] unittest for NDB_Config 

### PR DESCRIPTION
**Pull request by kongtiaowang**
Add a new unit test for NDB_Config.

### CODE REVIEW

File: test/unittests/VisitTest.php [hunk 1]
```diff
@@ -1,106 +0,0 @@
-<?php
```
Comment (driusan) [inline test/unittests/VisitTest.php]: Why does this PR remove this other test?

Comment (kongtiaowang) [reply test/unittests/VisitTest.php]: moving back

File: test/unittests/NDB_ConfigTest.php [hunk 2]
```diff
@@ -0,0 +1,242 @@
+<?php
```
Comment (johnsaigle) [inline test/unittests/NDB_ConfigTest.php]:
```suggestion
<?php declare(strict_types=1);
```

File: test/unittests/NDB_ConfigTest.php [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+     *
+     * @return void
+     */
+    public function testGetSettingFromXML()
+    {
+        $this->_config->_settings = array('aaa' => array("bbb" => "test"));
+        $this->assertEquals("test", $this->_config->getSettingFromXML("bbb"));
+    }
+    /**
+     * Test getSetting() method. Passing an array, it will parse the value.
```
Comment (PapillonMcGill) [inline test/unittests/NDB_ConfigTest.php]:
```suggestion
     * Test getSettingFromXML() method.Giving an array,
     it should return the value associated to a key.
```

File: test/unittests/NDB_ConfigTest.php [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+     * @return void
+     */
+    public function testGetSetting()
+    {
+        $this->_config->_settings = array('aaa' => array("bbb" => "unittest"));
+        $this->assertEquals("unittest", $this->_config->getSetting("bbb"));
+
+    }
+    /**
+     * Test getProjectSettings() method. Passing a projecID will return
```
Comment (PapillonMcGill) [inline test/unittests/NDB_ConfigTest.php]:
```suggestion
     * Test getProjectSettings() method. Giving a projectID, it should 
     return an array containing the project information.
```

File: test/unittests/NDB_ConfigTest.php [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+                    'recruitmentTarget' => '100',
+                   );
+        $this->_dbMock->expects($this->once())
+            ->method('pselectRow')
+            ->willReturn($info);
+        $this->assertEquals($result, $this->_config->getProjectSettings(999));
+
+    }
+    /**
+     * Test getSubprojectSettings() method. Passing a projecID will return
```
Comment (PapillonMcGill) [inline test/unittests/NDB_ConfigTest.php]:
```suggestion
     * Test getSubprojectSettings() method. Giving a projectID, it should
     return an array containing the subproject information.
```

File: test/unittests/NDB_ConfigTest.php [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+                   );
+
+        $this->_dbMock->expects($this->once())
+            ->method('pselectRow')
+            ->willReturn($info1);
+        $this->assertEquals($result, $this->_config->getSubprojectSettings(999));
+
+    }
+    /**
+     * Test getSubprojectSettings() method. Passing a projecID will return
```
Comment (PapillonMcGill) [inline test/unittests/NDB_ConfigTest.php]:
```suggestion
     * Test getSubprojectSettings() method. Giving a projectID, it should
     return an array containing the subproject information.
     Giving an invalid ID, it should return an empty array.
```

File: test/unittests/NDB_ConfigTest.php [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+     */
+    public function testGetSubprojectSettingsWithFakeID()
+    {
+        $this->_dbMock->expects($this->once())
+            ->method('pselectRow')
+            ->willReturn(null);
+        $this->assertEquals(array(), $this->_config->getSubprojectSettings(111));
+    }
+    /**
+     * Test getExternalLinks() method.Passing a valid ExternalLink, it will return an array.
```
Comment (PapillonMcGill) [inline test/unittests/NDB_ConfigTest.php]:
```suggestion
     * Test getExternalLinks() method.Giving a valid ExternalLink, it should
     return an array containing the URL associated with the link text.
```

File: test/unittests/NDB_ConfigTest.php [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+                   );
+
+        $this->_dbMock->expects($this->once())
+            ->method('pselectRow')
+            ->willReturn($info1);
+        $this->assertEquals($result, $this->_config->getSubprojectSettings(999));
+
+    }
+    /**
+     * Test getExternalLinks() method.Giving a valid ExternalLink, it should
```
Comment (PapillonMcGill) [inline test/unittests/NDB_ConfigTest.php]:
```suggestion
* Test getSubprojectSettings() method. Giving a projectID, it should
     return an array containing the subproject information.
     Giving an invalid ID, it should return an empty array.
```

**Pull request merged**